### DISCUSSION TITLE
MDX2: Can't resolve '@mdx-js/react' for Web component

### DISCUSSION THREAD

**Discussion by sanketzad** [storybookjs/storybook#20857] [Help]
<img width="983" alt="image" src="https://user-images.githubusercontent.com/40885376/215763786-89170859-9d79-4f40-8ec7-2a84744085a8.png">

Hello guys,
I was trying to update the Storybook to the latest version @6.5.16. When I was trying to start the Storybook, I was getting errors as shown in the image. 

## package.json: 

"devDependencies": {
    "@babel/core": "^7.20.12",
    "@babel/preset-react": "^7.18.6",
    "@storybook/addon-actions": "^6.5.16",
    "@storybook/addon-essentials": "^6.5.16",
    "@storybook/addon-links": "^6.5.16",
    "@storybook/builder-webpack4": "^6.5.16",
    "@storybook/builder-webpack5": "^6.5.16",
    "@storybook/manager-webpack4": "^6.5.16",
    "@storybook/manager-webpack5": "^6.5.16",
    "@storybook/web-components": "^6.5.16",
    "babel-loader": "9.1.2",
    "lit-html": "^2.6.1"
  },

## I tried installing the package from the update to @6.5.16, below were the warnings:

ERESOLVE overriding peer dependency
npm WARN While resolving: react-inspector@5.1.1
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peerOptional react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @storybook/addon-actions@6.5.16
npm WARN   node_modules/@storybook/addon-actions
npm WARN     dev @storybook/addon-actions@"^6.5.16" from the root project
npm WARN     1 more (@storybook/addon-essentials)
npm WARN   27 more (@storybook/addon-backgrounds, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.8.4 || ^17.0.0" from react-inspector@5.1.1
npm WARN node_modules/@storybook/addon-actions/node_modules/react-inspector
npm WARN   react-inspector@"^5.1.0" from @storybook/addon-actions@6.5.16
npm WARN   node_modules/@storybook/addon-actions
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.8.4 || ^17.0.0" from react-inspector@5.1.1
npm WARN   node_modules/@storybook/addon-actions/node_modules/react-inspector
npm WARN     react-inspector@"^5.1.0" from @storybook/addon-actions@6.5.16
npm WARN     node_modules/@storybook/addon-actions
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: @mdx-js/react@1.6.22
npm WARN Found: react@18.2.0
npm WARN node_modules/react
npm WARN   peerOptional react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @storybook/addon-actions@6.5.16
npm WARN   node_modules/@storybook/addon-actions
npm WARN     dev @storybook/addon-actions@"^6.5.16" from the root project
npm WARN     1 more (@storybook/addon-essentials)
npm WARN   27 more (@storybook/addon-backgrounds, ...)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react@"^16.13.1 || ^17.0.0" from @mdx-js/react@1.6.22
npm WARN node_modules/@storybook/addon-docs/node_modules/@mdx-js/react
npm WARN   @mdx-js/react@"^1.6.22" from @storybook/addon-docs@6.5.16
npm WARN   node_modules/@storybook/addon-docs
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^16.13.1 || ^17.0.0" from @mdx-js/react@1.6.22
npm WARN   node_modules/@storybook/addon-docs/node_modules/@mdx-js/react
npm WARN     @mdx-js/react@"^1.6.22" from @storybook/addon-docs@6.5.16
npm WARN     node_modules/@storybook/addon-docs

## When tried starting Storybook local server, below is the error:

ModuleNotFoundError: Module not found: Error: Can't resolve '@mdx-js/react' in '/Users/I575212/Documents/Workspace/components-storybook/stories'
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/lib/Compilation.js:2016:28
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/lib/NormalModuleFactory.js:798:13
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:10:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/lib/NormalModuleFactory.js:270:22
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:9:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/lib/NormalModuleFactory.js:434:22
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/lib/NormalModuleFactory.js:116:11
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/lib/NormalModuleFactory.js:670:25
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/lib/NormalModuleFactory.js:855:8
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/lib/NormalModuleFactory.js:975:5
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/neo-async/async.js:6883:13
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/webpack/lib/NormalModuleFactory.js:958:45
    at finishWithoutResolve (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:312:11)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:386:15
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:87:43
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/forEachBail.js:16:12
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/AliasPlugin.js:125:14
    at next (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/forEachBail.js:14:3)
    at forEachBail (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/forEachBail.js:24:9)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/AliasPlugin.js:51:5
    at _next0 (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:8:1)
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:30:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:27:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/DescriptionFilePlugin.js:87:43
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:16:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/Resolver.js:435:5
    at eval (eval at create (/Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/node_modules/tapable/lib/HookCodeFactory.js:33:10), <anonymous>:15:1)
    at /Users/I575212/Documents/Workspace/components-storybook/node_modules/enhanced-resolve/lib/DirectoryExistsPlugin.js:41:15
    at processTicksAndRejections (node:internal/process/task_queues:82:21)
resolve '@mdx-js/react' in '/Users/I575212/Documents/Workspace/components-storybook/stories'
  Parsed request is a module
  using description file: /Users/I575212/Documents/Workspace/components-storybook/package.json (relative path: ./stories)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module

Can anyone help me by providing a solution for the above issue mentioned? 

Thank you in advance.

**Comment by bzh2610** [comment]
React is now a required peer dependency of storybook in v7 : 

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#react-peer-dependencies-required

You could do :
`npm add react react-dom --dev`


But with v6, it should not be required.
Could you provide a more complete version of your package.json ?

> I tried installing the package from the update to @6.5.16

 Did you use `npx storybook upgrade` ?

From the logs, `Conflicting peer dependency` it appears that there is a conflict between some dependencies requirements, could you try to re-install the storybook dependencies with the `--legacy-peer-deps` flag ?

**Reply by sanketzad** [reply]
Hey @bzh2610 , thank you for the input.

Regarding package.json - whatever I have shared as devDependencies are the ones I am currently using it. There is no change in it. Do you mean, you need the full version package.json to investigate?

No, I didn't use ```npx storybook upgrade```, instead I used ```npx npm-check -uE``` to detect the latest version and then I updated the storybook to 6.5.16.

Yes, there are some ```Conflicting peer dependencies ```. So I tried with ```npm i --legecy-peer-deps```, still no luck because I was continuously getting error in ```postcss version error```.

**Reply by bzh2610** [reply]
Sorry for the delay, according to the output you posted before (npm install), it appears that your project has a dependency to `react@18.2.0` in the package.json, which is not compatible with packages `react-inspector@5.1.1` and `@mdx-js/react@1.6.22` and causing the dependency warning; could you try to downgrade your react version to 17 ?

Thus, I don't think this will fix the issue, sometimes mainteners just forget to say that the package is compatible with the latest version and no code-change is required within the library.


I don't know how `npx npm-check -uE` works but it might just update the dependencies to the latest version without running the additions migration script.

Could you try again with `npx storybook upgrade` and let me know what is the output ?
